pkg/operator/reports.go

```diff
@@ -354,38 +354,68 @@ func getReportPeriod(now time.Time, logger log.FieldLogger, report *metering.Rep
 	return reportPeriod, nil
 }
 
+// handleExpiredReport is responsible for determining if the @report object we're
+// currently processing has reached it's configured retention period. In the case
+// where the Report object hasn't been configured with an expiration, exit early,
+// else attempt to determine if we can safely delete this object without breaking
+// other Report custom resources that rely on this particular Report.
 func (op *Reporting) handleExpiredReport(logger log.FieldLogger, report *metering.Report, now time.Time) error {
-	// check if the Report is being deleted already
+	// check if the Report deletion timestamp has already been set, exit early.
 	if report.DeletionTimestamp != nil {
-		logger.Warnf("report was already marked for deletion")
+		// TODO: should we be requeuing here?
```

Comment:
We can leave this here for now, but I assume we should be re-queueing the resource here instead of logging and exiting early.